data/blockchain/blockchain.go

```diff
@@ -19,12 +20,14 @@ type BlockChain struct {
 	localHeight            int64          // Height of the local chain
 	networkHeight          int64          // Percieved height of the network chain
 	badBlocks              storage.Cacher // Bad blocks cache
+	appStatusHandler       core.AppStatusHandler
 }
 
 // NewBlockChain returns an initialized blockchain
 // It uses a config file to setup it's supported storage units map
 func NewBlockChain(
 	badBlocksCache storage.Cacher,
+	appStatusHandler core.AppStatusHandler,
```

Comment:
This should not be injected as it is an auxiliary struct/functionality. Should be automatically instantiated with NilAppStatusHandler and the Blockchain struct should have a Set function. Apply this here, block processor. Node already has an With... func so it is ok there. 
Other "check for nils" comments should apply on the set func.

Reply:
because some things need to be updated when an event occurs. For example, 'IsSyncing' should be updated instantly